sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ImplementIDisposableCorrectly.cs

```diff
@@ -286,6 +287,10 @@ private bool HasVirtualDisposeBool(INamedTypeSymbol typeSymbol)
                     .WhereNotNull()
                     .Any(methodSym => !methodSym.IsAbstract);
             }
+
+            private static bool HasBodyOrExpressionBody(BaseMethodDeclarationSyntax methodDeclaration) =>
+                methodDeclaration != null &&

```

Comment:
It's ok like this but you could have use null-conditional on both path and forget about the first AND